community/io/src/test/java/org/neo4j/graphdb/mockfs/EphemeralFileSystemAbstraction.java

```diff
@@ -1199,18 +1206,22 @@ private ByteBuffer allocate( int sizeIndex )
 
         void free()
         {
+            assertNotFreed();
+
             try
             {
                 clear();
             }
             finally
             {
                 buf = null;
+                freeCall = new Exception();
```

Comment:
I'd add a descriptive message to this exception so it's easy to recognise in the stack traces what these stack frames mean.